### PR TITLE
Add sectionC to the state system for the new charm search area

### PR DESCRIPTION
**Pull request by hatched**

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2899/
Test FAILed.

**Comment by hatched** [comment]
CI server was out of space - manually ran passed http://ci.jujugui.org:8080/job/juju-gui/2900/

File: jujugui/static/gui/src/app/views/state.js [hunk 1]
```diff
@@ -144,7 +144,8 @@ YUI.add('juju-app-state', function(Y) {
       var sections = {
         app: 'App',
         sectionA: 'SectionA',
-        sectionB: 'SectionB'
+        sectionB: 'SectionB',
+        sectionC: 'SectionC'
       };
       // calls _dispatchSectionA or _dispatchSectionB
```
Comment (makyo) [inline jujugui/static/gui/src/app/views/state.js]: "calls dispatch with the appropriate section" maybe?

File: jujugui/static/gui/src/app/views/state.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ YUI.add('juju-app-state', function(Y) {
       @param {Object} state The current state object.
     */
     dispatch: function(state) {
-      var sections = ['app', 'sectionA', 'sectionB'];
+      var sections = ['app', 'sectionA', 'sectionB', 'sectionC'];
```
Comment (makyo) [inline jujugui/static/gui/src/app/views/state.js]: I still really dislike having these be named, and also to be named twice, but that's out of scope.

**Comment by makyo** [comment]
:+1: qaing now

File: jujugui/static/gui/src/app/views/state.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/views/state.js]: yes I have a task to allow dynamic registering of sections...but this was a quick and dirty way to add this support for now

File: jujugui/static/gui/src/app/views/state.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/views/state.js]: sure I can update that, was just an oversight when doing all these mechanical changes.

**Comment by kadams54** [comment]
:+1: for me as well.

**Comment by makyo** [comment]
QA okay

**Comment by hatched** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**